### PR TITLE
Footer

### PR DESCRIPTION
**Pull request by cit280superepicness**

### CODE REVIEW

**Comment by cit280superepicness** [comment]
Not merged.

**Comment by cit280superepicness** [comment]
Closed by mistake, Reopened

**Comment by cit280superepicness** [comment]
Sorry. I know I added the code to check for an email that already exists when a user registers and fixed the clear button on the Register page in the wrong branch.

File: src/main/resources/templates/register.html [hunk 1]
```diff
@@ -72,6 +73,25 @@
             </div>
         </div>
     </div>
+
+    <script>
+    // clear form data
+    function clearText(){
```
Comment (jrmullen) [inline src/main/resources/templates/register.html:79]: What makes this different than our pre-existing clear button?

File: src/main/java/com/watchdog/dao/user/UserDaoImpl.java [hunk 2]
```diff
@@ -94,8 +115,6 @@ public void update(User user) {
     @Override
     public void deleteById(int id) {
 
-        String query = "delete from user where USER_ID=?";
```
Comment (jrmullen) [inline src/main/java/com/watchdog/dao/user/UserDaoImpl.java:97]: Good catch

File: src/main/java/com/watchdog/dao/Constants.java [hunk 3]
```diff
@@ -12,6 +12,7 @@
     public static final String DELETE_USER_BY_ID_QUERY = "delete from user where USER_ID=?";
     public static final String GET_ALL_USERS_QUERY = "select USER_ID, USER_FNAME, USER_LNAME, USER_EMAIL from user";
     public static final String GET_BY_EMAIL_QUERY = "select USER_FNAME, USER_LNAME, USER_ID from user where USER_EMAIL = ?";
```
Comment (jrmullen) [inline src/main/java/com/watchdog/dao/Constants.java:14]: Not part of your pull request, but I noticed it. Refactor GET_BY_EMAIL_QUERY into GET_USER_BY_EMAIL_QUERY

File: src/main/java/com/watchdog/dao/Constants.java [hunk 4]
```diff
@@ -12,6 +12,7 @@
...
+    public static final String GET_EMAIL_BY_EMAIL_QUERY = "select USER_EMAIL from user where USER_EMAIL = ?";
```
Comment (jrmullen) [inline src/main/java/com/watchdog/dao/Constants.java:15]: Do you think this constant name is acceptible? Seems a little strange to me but your call. I guess it technically kind of makes sense

GET_USER_EMAIL_QUERY?
GET_EMAIL_QUERY?
SELECT_EMAIL_QUERY?
or leave it how it is. Your call.

File: src/main/java/com/watchdog/dao/user/UserDaoImpl.java [hunk 5]
```diff
@@ -80,6 +80,27 @@ public User mapRow(ResultSet rs, int rowNum)
     }
 
     @Override
+    public boolean emailAlreadyExists(String email){
```
Comment (jrmullen) [inline src/main/java/com/watchdog/dao/user/UserDaoImpl.java:83]: I don't think this name is sufficient. I know it is a boolean T/F for whether or not the email address already exists, but method names are generally verbs. How about checkEmailExists or something along those lines? Or to make it even more clear what this method is trying to accomplish you could do something like checkIfEmailIsTaken or checkEmailTaken

Comment (jrmullen) [reply src/main/java/com/watchdog/dao/user/UserDaoImpl.java:83]: This also seems like a lot of code for such a simple query. 

```
@Override
public boolean emailAlreadyExists(String email){
    try {
        jdbcTemplate.queryForObject(Constants.GET_EMAIL_BY_EMAIL_QUERY, new Object[]{email}, String.class);
        return true;
    } catch(EmptyResultDataAccessException e) {
        return false;
    }
}
```

I haven't actually tried this yet, but I think something similar to this should suffice for what we are trying to accomplish. I'm pretty sure the rowMapper is used for objects with many attributes, like our Users which have ID, first name, last name, email, password, etc. In that case you need a row mapper if you are going to be returning an object containing all of those attributes. In this case, however, we only are querying a true/false for one email address so I don't think it is necessary to include all of that extra row mapping stuff.

File: src/main/java/com/watchdog/dao/user/UserDao.java [hunk 6]
```diff
@@ -16,7 +16,9 @@
     //Read
     User getById(int id);
 
-    User getByEmail(String id);
+    User getByEmail(String email);
```
Comment (jrmullen) [inline src/main/java/com/watchdog/dao/user/UserDao.java:19]: Good catch. I have no idea why this said id.

**Comment by cit280superepicness** [comment]
I am feeling better and will be in class Thursday and fix all of these
then. Thanks for being so thorough.

**Comment by cit280superepicness** [comment]
Yes, this is needed. Briana applied a similar fix on the device manager
page.

**Comment by cit280superepicness** [comment]
I don't think that should be there at all and no idea how it got there. Lol
I will fix it with what we had before.

**Comment by cit280superepicness** [comment]
I was thinking the same thing when I named it but didn't know what else to
call it based on our naming conventions.  If you have any ideas for an
alternative name, please let me know and I will be happy to change it.

**Comment by cit280superepicness** [comment]
I don't have time to look into an alt method for less lines of code here.
But if you know of one, feel free to fix it or let me know how and I will
change it. Yes. I agree the way I did this is a bit sloppy though code wise.

**Comment by jrmullen** [comment]
FYI Renee you have to reply (to a comment) for your comment to be attached or else it just goes as a comment on the whole pull request. And generally after you makes changes based on a comment in a pull request you comment done or fixed or something like that. None of these comments are attached to any of my comments.

Just so you know. It's good practice.

Unless of course you are replying from email maybe? In which case I highly recommend working with pull requests from the actual github pull request page.

**Comment by cit280superepicness** [comment]
Got it. I am going to add the comments I made to the appropriate place.

File: src/main/resources/templates/register.html (hunk 1 above)
Comment (cit280superepicness) [reply src/main/resources/templates/register.html:79]: Yes, this is needed. Briana applied a similar fix on the device manager
page. See the email I sent you about it.

File: src/main/java/com/watchdog/dao/user/UserDaoImpl.java (hunk 2 above)
Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/user/UserDaoImpl.java:97]: I don't think that should be there at all and no idea how it got there. Lol
I will fix it with what we had before.

File: src/main/java/com/watchdog/dao/Constants.java (hunk 4 above)
Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/Constants.java:15]: I was thinking the same thing when I named it but didn't know what else to
call it based on our naming conventions.  If you have any ideas for an
alternative name, please let me know and I will be happy to change it.

File: src/main/java/com/watchdog/dao/user/UserDaoImpl.java (hunk 5 above)
Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/user/UserDaoImpl.java:83]: I don't have time to look into an alternate method for less lines of code here.
But if you know of one, feel free to fix it or let me know how and I will
change it. Yes. I agree the way I did this is a bit sloppy though code wise.

File: src/main/java/com/watchdog/dao/user/UserDaoImpl.java (hunk 2 above)
Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/user/UserDaoImpl.java:97]: Nm, wrong comment I'm responding to

File: src/main/java/com/watchdog/dao/Constants.java (hunk 4 above)
Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/Constants.java:15]: Nm about not knowing what to rename it, I will rename it to something you suggested. On my email, your whole comment didn't show for some reason.

Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/Constants.java:15]: Fixed.

File: src/main/java/com/watchdog/dao/user/UserDaoImpl.java (hunk 5 above)
Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/user/UserDaoImpl.java:83]: Renamed "emailAlreadyExists" to "checkEmailExists".

File: src/main/java/com/watchdog/dao/Constants.java (hunk 3 above)
Comment (cit280superepicness) [reply src/main/java/com/watchdog/dao/Constants.java:14]: Fixed

**Pull request merged**